### PR TITLE
(PUP-10597) Remove deprecated held

### PR DESCRIPTION
**Pull request by Dorin-Pleava**

### CODE REVIEW

**Comment by gimmyxd** [comment]
@Dorin-Pleava we should entirely remove "held" from ensure values, as it was replaced by apt mark: https://github.com/puppetlabs/puppet/pull/7802

**Comment by gimmyxd** [comment]
Jenkins please test this on debian8-64,debian9-64,debian10-64

**Comment by Dorin-Pleava** [comment]
Jenkins please test this on debian10-64a,debian8-64a,debian8-32a,debian9-64a,debian9-32a

**Comment by gimmyxd** [comment]
Jenkins please test this on debian9-64,debian10-64

File: acceptance/tests/provider/package/dpkg_hold_true_package_is_latest.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-test_name "dpkg ensure held package is latest installed" do
+test_name "dpkg ensure hold package is latest installed" do
   confine :to, :platform => /debian-8-amd64/
```
Comment (gimmyxd) [inline acceptance/tests/provider/package/dpkg_hold_true_package_is_latest.rb:2]: can you please update this to debian-9 as debian-8 was dropped in puppet-agent main

File: acceptance/tests/provider/package/dpkg_hold_true_should_preserve_version.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-test_name "dpkg ensure held package should preserve version if package is already installed" do
+test_name "dpkg ensure hold package should preserve version if package is already installed" do
   confine :to, :platform => /debian-8-amd64/
```
Comment (gimmyxd) [inline acceptance/tests/provider/package/dpkg_hold_true_should_preserve_version.rb:2]: can you please update this to debian-9 as debian-8 was dropped in puppet-agent main

File: lib/puppet/provider/package/dpkg.rb [hunk 3]
```diff
@@ -173,14 +173,10 @@ def purge
...
 
-  def deprecated_hold
+  def hold
     if package_not_installed?
       self.install
     end
-    hold
-  end
-
-  def hold
```
Comment (ciprianbadescu) [inline lib/puppet/provider/package/dpkg.rb:183]: I'm not sure we still need to call self.install in this step, as it may run the install twice

**Comment by GabrielNagy** [comment]
I believe this works okay, but while testing I found a behavior that's not idempotent (also happens without changes from this PR).

The following will always report changes on Debian 10:

```puppet
package { 'mc':
    ensure => present,
    mark   => none,
}
```

/cc @ciprianbadescu

File: lib/puppet/provider/package/dpkg.rb (hunk 3 above)
Comment (Dorin-Pleava) [reply lib/puppet/provider/package/dpkg.rb:183]: Yep, it seems it installs the package without doing the `self.install` here. 
I think `package_not _installed?` method can also be removed as I only found it being used here.

**Comment by ciprianbadescu** [comment]
> package { 'mc': ensure => present, mark => none, }

Should be solved with bellow change:
```
--- a/lib/puppet/provider/package/dpkg.rb
+++ b/lib/puppet/provider/package/dpkg.rb
@@ -74,7 +74,7 @@ Puppet::Type.type(:package).provide :dpkg, :parent => Puppet::Provider::Package
       elsif ['config-files', 'half-installed', 'unpacked', 'half-configured'].include?(hash[:status])
         hash[:ensure] = :absent
       end
-      hash[:mark] = :hold if hash[:desired] == 'hold'
+      hash[:mark] = hash[:desired] == 'hold' ? :hold : :none
     else
       Puppet.debug("Failed to match dpkg-query line #{line.inspect}")
     end

```

**Review by ciprianbadescu — APPROVED** [review]

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**